sh_scrapy/extension.py

```diff
@@ -48,13 +49,24 @@ def spider_closed(self, spider, reason):
         self.pipe_writer.set_outcome(reason)
 
 
+_DOWNLOADER_MIDDLEWARE_WARNING = """\
+sh_scrapy.extension.HubstorageMiddleware functionality is now split between two new middlewares:
+
+- sh_scrapy.middlewares.HubstorageDownloaderMiddleware
+- sh_scrapy.middlewares.HubstorageSpiderMiddleware
+
+Please migrate to new middlewares.
+"""
+
+
 class HubstorageMiddleware(object):
 
     def __init__(self):
+        warnings.warn(_DOWNLOADER_MIDDLEWARE_WARNING)
```

Comment:
@dangra what if someone subclasses it? if we remove it it'll break the code, so I decided to keep it. I've never seen such cases but better to be on the safe side and issue a warning on init instead of import error.

regards util for class deprectation - it's useful when you want to move class to a different location and it requires to specify new class. here we don't have one-to-one old-new relation, it doesn't make a lot of sense to instantiate any of the new classes instead of the old one. also new classes don't have any